### DISCUSSION TITLE
Help with NetCDF file and MODE

### DISCUSSION THREAD

**Discussion by balbrightwpc** [dtcenter/METplus#1347] [File I/O] [answered]
I've received a NetCDF file from a partner with 24 hour accumulated
snowfall in mm and I am trying to run MODE using a NOHRSC grib2 file. When
running MODE I get the following error:

ERROR  :
ERROR  : Trouble reading forecast file
"/export/hpc-lw-hmtdev2/balbright/Desktop/python/wwe2022/mode/archive/20211231/GFSDS_20211231_WPC_0000_006.nc"
ERROR  :

The NetCDF file can be downloaded here:
https://www.inscc.utah.edu/~u0813947/exp_downscaled_slr/example_NetCDF/

I'm not really sure the easiest way to show the script I am working on or the observation dataset file I used via the Github forum. I had emailed met_help and it redirected me here. Any help you could provide as to why MODE might not be able to read the NetCDF file would be appreciated. I am not sure if it is formatted correctly for MODE or if there is some other issue. Thank you.

**Comment by jprestop** [comment, marked as answer]
Hi @balbrightwpc. 

MET can read a few different flavors of NetCDF files:

- its internal format which is output from tools like gen_vx_mask or pcp_combine
- files following the [Climate Forecast (CF) convention](https://cfconventions.org/latest.html)
- WRF output post-processed by the WRF-interp utility

It appears that GFSDS_20211231_WPC_0000_006.nc is an attempt to follow the internal NetCDF file format used by MET.  We strongly recommend that users follow the NetCDF Climate-Forecast convention rather than the custom MET NetCDF format. Eventually, we hope to switch MET over to follow the CF convention as well.

Running this data through plot_data_plane, as is, using "file_type = NETCDF_MET;" produces partial success:
```
plot_data_plane GFSDS_20211231_WPC_0000_006.nc apcp.ps 'name="APCP";level="(*,*)"; file_type=NETCDF_MET;' -v 4
```
```
WARNING: read_netcdf_grid() -> Applying METv2.0 grid parsing logic since the "MET_version" global attribute is not present.
WARNING: The field "APCP(*,*)" contains no valid data!
```
While it runs without error, the output is *bad*. The "MET_version" global attribute is what triggers the right library logic in the MET library code.  Manually adding that global attribute and rerunning, only leads to the next problem:
```
ncatted -a MET_version,global,c,c,"V10.0.0" GFSDS_20211231_WPC_0000_006.nc -o GFSDS_20211231_WPC_0000_006_MOD.nc
```
```
plot_data_plane GFSDS_20211231_WPC_0000_006_MOD.nc apcp.ps 'name="APCP"; level="(*,*)";' -v 4
```
```
ERROR  : plot_data_plane -> trouble getting field "name="APCP"; level="(*,*)";" from file "GFSDS_20211231_WPC_0000_006_MOD.nc"
```
The grid definition is missing from this file. Judging by the dimensions of the lat and lon variables:

float lat(lat, lon) ;
lat:units = "degrees_north" ;
lat:standard_name = "latitude" ;
lat:_CoordinateAxisType = "Lat" ;
float longitude(lat, lon) ;
longitude:units = "degrees_east" ;
longitude:standard_name = "longitude" ;
longitude:_CoordinateAxisType = "Lon

This is just simple lat, lon data. The "longitude" variable has 2 dimensions. And looking at it in ncview, we see that the values vary over the domain. So we suspect this data is actually projected. So we'd need the grid definition for this domain added to the file.

But again, we'd recommend shifting gears and switching to follow the NetCDF CF-conventions instead, if possible.

**Reply by jprestop** [reply]
Hi @balbrightwpc.  I just wanted to follow up to see if you had any follow up questions on this Discussion?  I'd like to mark this discussion as being answered and "lock it" to prevent future posts. That's how we encourage users to create new discussions for new questions. But I wanted to give you an opportunity to comment on it before doing so.

**Reply by balbrightwpc** [reply]
Hi @jprestop, feel free to lock the topic. Your information was very helpful and my partner was able to get the files formatted properly based on the information you provided. Thank you very much!

**Reply by jprestop** [reply]
Great! Thanks so much for the update!
